server/src/services/formatting/forgeFormat.ts

```diff
@@ -3,23 +3,33 @@ import path from "path";
 import { TextDocument } from "vscode-languageserver-textdocument";
 import { TextEdit } from "vscode-languageserver-types";
 import { URI } from "vscode-uri";
+import { promisify } from "util";
 import { resolveForgeCommand } from "../../frameworks/Foundry/resolveForgeCommand";
 
 export async function forgeFormat(text: string, document: TextDocument) {
   const forgeCommand = await resolveForgeCommand();
-
   const documentPath = URI.parse(document.uri).fsPath;
-
   const cwd = path.dirname(documentPath);
 
-  const formattedText = cp
-    .execSync(`${forgeCommand} fmt --raw -`, {
-      cwd,
-      input: text,
-      stdio: "pipe",
-    })
-    .toString();
+  // Spawn and promisify the forge fmt process
+  const execPromise = promisify(cp.exec)(`${forgeCommand} fmt --raw -`, {
+    cwd,
+  });
+
+  // Write the file contents to the process
+  const { child } = execPromise;
+  child.stdin?.write(text);
+  child.stdin?.end();
+
+  // Set a time limit
+  setTimeout(() => {
```

Comment:
Can we cancel this timeout after the `await execPromise`?

I don't think it is a memory leak, I just don't want kills on closed child processes.